feat(logs): Implement load more functionality for logs with pagination

jonmcwest opened the pull request:
## Problem

You couldn't view all the logs that match the current filters, only the first 100

## Changes

<img width="1225" height="154" alt="image" src="https://github.com/user-attachments/assets/a708c261-5808-4bb2-bd66-9c19c833b72e" />

- Adds "Load more" button to load more logs

## How did you test this code?

Local Dev

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

jonmcwest committed:
feat(logs): Implement load more functionality for logs with pagination

github-actions[bot] committed:
test(storybook): update UI snapshots

jonmcwest committed:
refactor(logs): only fetch logs on loadmorelogs

jonmcwest commented on products/logs/frontend/logsLogic.tsx:
```diff
@@ -250,8 +264,7 @@ export const logsLogic = kea<logsLogicType>([
                         query: {
-                            limit: 100,
-                            offset: values.logs.length,
+                            limit: values.logsLimit,
```
This is ok for the moment, quick fix for now.

benjackwhite commented on products/logs/frontend/logsLogic.tsx:
```diff
@@ -250,8 +264,7 @@ export const logsLogic = kea<logsLogicType>([
                         query: {
-                            limit: 100,
-                            offset: values.logs.length,
+                            limit: values.logsLimit,
```
This feels... not good. We're just going to hit the same endpoint with more and more entries??

I think you might want to use workflow logs as a reference point and implement timestamp based loading which will be way more effective https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/hog-functions/logs/logsViewerLogic.tsx

jonmcwest committed:
refactor(logs): remove redundant condition

jonmcwest committed:
refactor(logs): implement more efficient timestamp based pagination for logs

- Updated API response to include `hasMore` for pagination support.
- Adjusted logs fetching logic to handle pagination correctly in the frontend.
- Added `parseLogAttributes` function to ensure log attributes are properly formatted.
- Introduced `clearLogs` action to reset logs state on filter change
- Updated logic to determine if more logs can be loaded based on the new `hasMore` flag.

jonmcwest committed:
refactor(logs): remove redundant selector

github-actions[bot] committed:
Update query snapshots

jonmcwest committed:
Merge branch 'master' into jon/logs-limits
